Whitelisting.py

```diff
@@ -119,11 +121,19 @@ def compareDigests(self, pkg):
             try:
                 h = hashlib.new(alg)
 
-                # NOTE: this path is dynamic and rpmlint unpacks the RPM
+                info = fileinfos[path]
+
+                # walk through symlinks within the package, otherwise opening
+                # them directly will cause errors or undefined results,
+                # because we'll be looking up system paths
+                while info.linkto:
+                    info = fileinfos[info.linkto]
```

Comment:
This will raise an exception if a symlink points to a file that isn't part of the same RPM. (It also doesn't cover relative symlinks. IIRC there might be some upcoming changes to switch our RPMs to those.)

Reply:
If a symlink points outside of the RPM it's fine for me if an exception is thrown. This location is only reached if a whitelisting entry exists for the path, and if one exists then the target should we within the same RPM, the outside of the RPM case is not supported by the whitelisting mechanism and I don't see how we could support it.

Are you sure it dousn't cover relative symlinks? The rpmlint data structure performs some magic here, I'll have a look what happens when relative symlinks occur.

Another issue I thought of is that is could produce an infinite loop if a crafted RPM occurs. But of these DoS scenarios probably hundreds exist in checkers.

Reply:
So I'm now using the readlink function provided by the rpmlint `Pkg` class. This simplifies things a lot as it handles both relative and absolute symlinks and stays within the RPM tree. It is not robust against symlink loops, however. I'm willing to accept that, maybe a fix in rpmlint upstream is in order for this.

Reply:
Nice find :). Infinite loop seems 'safe' - people can do that in `%build` directly, so it doesn't matter if `rpmlint` can be tricked into one.